### PR TITLE
Allow attaching TTYs to containers

### PR DESCRIPTION
**Pull request by rgulewich**
This exposes a new passthrough parameter `titusParameter.agent.ttyEnabled`, which causes the container to be created with a tty (same as doing a `docker run --tty`).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -140,5 +140,5 @@ $(GOVENDOR):
 .PHONY: testdeps
 testdeps: $(GOVENDOR)
 	$(GOVENDOR) install +local
-	# Fail if gometalinter is not present in $PATH:
+	# Fail if gometalinter is not present in PATH:
```
Comment (sargun) [inline Makefile]: Wanna break this out into another PR / commit?

File: README.md [hunk 2]
```diff
@@ -46,6 +46,16 @@ If you want to build a dpkg, without Docker, once the code is checked out, you c
 make build-standalone
 ```
 
+## Linting
+Linting is done via the [gometalinter](https://github.com/alecthomas/gometalinter) package, which runs various linters. In order to ensure
```
Comment (sargun) [inline README.md]: See above.

File: README.md [hunk 3]
```diff
@@ -46,6 +46,16 @@ If you want to build a dpkg, without Docker, once the code is checked out, you c
...
+consistent versions of linters, install version 2.0.11 as per the instructions in "Initial setup steps" above.
+
+To run lint checks:
+```sh-session
+make validate
```
Comment (sargun) [inline README.md]: make metalinter FAST=1 should probably be a thing.

File: executor/mock/standalone/standalone_test.go [hunk 4]
```diff
@@ -847,3 +848,16 @@ func testTwoCPUs(t *testing.T, jobID string) {
 		t.Fail()
 	}
 }
+
+func testTty(t *testing.T, jobID string) {
+	ji := &mock.JobInput{
+		ImageName:     ubuntu.name,
+		Version:       ubuntu.tag,
+		EntrypointOld: `/usr/bin/tty`,
```
Comment (sargun) [inline executor/mock/standalone/standalone_test.go]: Does this test to see if TTY is enabled (does it fail if TTY is disabled, if so, does it make sense to add a negative test?)

File: executor/mock/jobrunner.go [hunk 5]
```diff
@@ -201,14 +203,21 @@ func (jobRunner *JobRunner) StopExecutorAsync() {
...
 	var jobID string
+	attrs := make(map[string]string)
+
 	if jobInput.JobID == "" {
 		jobID = fmt.Sprintf("Titus-%v%v", r.Intn(1000), time.Now().Second())
 	} else {
 		jobID = jobInput.JobID
 	}
+
+	if jobInput.Tty {
```
Comment (sargun) [inline executor/mock/jobrunner.go]: Why don't you put this after creating titus.ContainerInfo? kinda like how this gets set: https://github.com/Netflix/titus-executor/blob/master/executor/mock/jobrunner.go#L249-L251

File: Makefile (hunk 1 above)
Comment (rgulewich) [reply Makefile]: They're already separate commits, but I can break them into another PR, sure.

File: executor/mock/standalone/standalone_test.go (hunk 4 above)
Comment (rgulewich) [reply executor/mock/standalone/standalone_test.go]: Yeah - `tty` will fail if there's no tty present.  I can add a negative test.

File: executor/mock/jobrunner.go (hunk 5 above)
Comment (rgulewich) [reply executor/mock/jobrunner.go]: Good point - missed that.  Done.
